library/include/rocwmma/internal/dpp_impl.hpp

```diff
@@ -103,13 +104,20 @@ namespace rocwmma
                           typename DataT>
                 ROCWMMA_DEVICE static inline DataT exec(DataT src0, DataT src1)
                 {
-                    reinterpret_cast<int32_t&>(src0) = __builtin_amdgcn_update_dpp(
-                        reinterpret_cast<int32_t const&>(src1), // fill value 'prev'
-                        reinterpret_cast<int32_t const&>(src0), // Src value
-                        DppCtrl::opCtrl(), // DPP control code
-                        WriteRowMask, // Mask for affected rows
-                        WriteBankMask, // Mask for affected banks
-                        BoundCtrl); // Fill in 0 on invalid indices
+                    constexpr size_t VecSize = sizeof(DataT) / sizeof(uint32_t);
+                    auto             dpp     = [](auto&& idx, auto&& v0, auto&& v1) {
+                        constexpr auto i = std::decay_t<decltype(idx)>::value;
+                        *(reinterpret_cast<uint32_t*>(&v0) + i) = __builtin_amdgcn_update_dpp(
+                            *(reinterpret_cast<uint32_t const*>(&v1) + i), // fill value 'prev'
+                            *(reinterpret_cast<uint32_t const*>(&v0) + i), // Src value
+                            DppCtrl::opCtrl(), // DPP control code
+                            WriteRowMask, // Mask for affected rows
+                            WriteBankMask, // Mask for affected banks
+                            BoundCtrl); // Fill in 0 on invalid indices
+                        return 0; // discard return value
```

Comment:
This is a bit misleading to return 0 here. Was there an extra copy or something here we are trying to remove?
src0 is definitely a by-copy argument, but return value elision should be able to avoid the extra copy.

Reply:
changed to return the result of cross-lane instructions.